test/unit/init/CliRunner.test.ts

```diff
@@ -0,0 +1,104 @@
+import * as Path from 'path';
+import type { Loader } from 'componentsjs';
+import { runCli } from '../../../src/init/CliRunner';
+import type { Setup } from '../../../src/init/Setup';
+import type { Logger } from '../../../src/logging/Logger';
+
+let calledInstantiateFromUrl: boolean;
+let calledRegisterAvailableModuleResources: boolean;
+let throwError: boolean;
+let outsideResolve: () => void;
+let functionToResolve: Promise<unknown>;
+
+const mockSetup = {
+  setup: jest.fn(),
+} as unknown as jest.Mocked<Setup>;
+
+// Mock the Loader class.
+jest.mock('componentsjs', (): any => (
+  // eslint-disable-next-line @typescript-eslint/naming-convention, object-shorthand
+  { Loader: function(): Loader {
+    return {
+      instantiateFromUrl(): any {
+        calledInstantiateFromUrl = true;
+        if (throwError) {
+          throw new Error('Error! :o');
+        }
+        return mockSetup;
+      },
+      registerAvailableModuleResources(): any {
+        calledRegisterAvailableModuleResources = true;
+      },
+    } as unknown as Loader;
+  } }
+));
+
+// Create a mock for the Logger as this doesn't exist in a testing environment.
+jest.mock('../../../src/logging/LogUtil', (): any => ({
+  getLoggerFor(): Logger {
+    return {
+      info(): any {
+        // The info method will be called when all other code has been executed, so end the waiting function.
+        outsideResolve();
+      },
+    } as unknown as Logger;
+  },
+}));
```

Comment:
Having another look at this I realize it's probably not a good idea to depend on a logger line for making sure everything works in order. Would it also work if you add the `outsideResolve()` to the `setup.setup()` call instead? Probably a bit more robust since logger calls are the most likely ones to change. 

Regarding actually making the logger not crash there is the line `setGlobalLoggerFactory(new VoidLoggerFactory());` you can use. You can do a search through the code to see where this is used in other tests.

If this does not work for the logger or no longer has full coverage afterwards you can ignore this.